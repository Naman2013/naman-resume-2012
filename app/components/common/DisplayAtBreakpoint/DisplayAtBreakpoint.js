import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import style from './DisplayAtBreakpoint.style';

const DisplayAtBreakpoint = ({
  children,
  screenSmall,
  screenMedium,
  screenLarge,
  screenXLarge,
}) => (
  <div className={classnames('root', {
    screenSmall,
    screenMedium,
    screenLarge,
    screenXLarge,
  })}
  >
    {children}
    <style jsx>{style}</style>
  </div>
);

DisplayAtBreakpoint.propTypes = {
  children: PropTypes.node.isRequired,
  screenSmall: PropTypes.bool,
  screenMedium: PropTypes.bool,
  screenLarge: PropTypes.bool,
  screenXLarge: PropTypes.bool,
};

DisplayAtBreakpoint.defaultProps = {
  screenSmall: false,
  screenMedium: false,
  screenLarge: false,
  screenXLarge: false,
};

export default DisplayAtBreakpoint;
