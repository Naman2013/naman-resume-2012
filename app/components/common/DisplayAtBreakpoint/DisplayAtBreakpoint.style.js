import css from 'styled-jsx/css';
import { screenMedium, screenLarge, screenXLarge } from 'styles/variables/breakpoints';

export default css`
  .root { display: none; width: 100%; }

  .root.screenSmall { display: block; }

  @media ${screenMedium} {
    .root.screenSmall { display: none; }
    .root.screenMedium { display: block; }
  }

  @media ${screenLarge} {
    .root.screenSmall { display: none; }
    .root.screenMedium { display: none; }
    .root.screenLarge { display: block; }
  }

  @media ${screenXLarge} {
    .root.screenSmall { display: none; }
    .root.screenMedium { display: none; }
    .root.screenLarge { display: none; }
    .root.screenXLarge { display: block; }
  }
`;
