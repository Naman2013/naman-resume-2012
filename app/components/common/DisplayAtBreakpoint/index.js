import DisplayAtBreakpoint from './DisplayAtBreakpoint';

export default DisplayAtBreakpoint;
