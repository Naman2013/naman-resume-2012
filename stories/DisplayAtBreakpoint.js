import React from 'react';
import { storiesOf } from '@storybook/react';
import DisplayAtBreakpoint from 'components/common/DisplayAtBreakpoint';
import { primaryFont } from 'styles/variables/fonts';

const TestComponent = ({ text }) => (
  <div style={{ background: 'blue', textAlign: 'center', padding: '20px' }}>
    <h1 style={{ color: 'white', fontFamily: primaryFont, fontSize: '16px' }}>{text}</h1>
  </div>
);

storiesOf(' DisplayAtBreakpoint', module)
  .add('Display at all breakpoints', () => (
    <DisplayAtBreakpoint
      screenSmall
      screenMedium
      screenLarge
      screenXLarge
    >
      <TestComponent text="Display at all breakpoints" />
    </DisplayAtBreakpoint>
  ))
  .add('Display small only', () => (
    <DisplayAtBreakpoint
      screenSmall
    >
      <TestComponent text="Screen is small" />
    </DisplayAtBreakpoint>
  ))
  .add('Display medium only', () => (
    <DisplayAtBreakpoint
      screenMedium
    >
      <TestComponent text="Screen is medium" />
    </DisplayAtBreakpoint>
  ))
  .add('Display large only', () => (
    <DisplayAtBreakpoint
      screenLarge
    >
      <TestComponent text="Screen is large" />
    </DisplayAtBreakpoint>
  ))
  .add('Display XLarge only', () => (
    <DisplayAtBreakpoint
      screenXLarge
    >
      <TestComponent text="Screen is XLarge" />
    </DisplayAtBreakpoint>
  ))
  .add('Display at small or large only', () => (
    <DisplayAtBreakpoint
      screenSmall
      screenLarge
    >
      <TestComponent text="Screen is either small or large" />
    </DisplayAtBreakpoint>
  ))
  .add('Display at medium or XLarge only', () => (
    <DisplayAtBreakpoint
      screenMedium
      screenXLarge
    >
      <TestComponent text="Screen is either medium or Xlarge" />
    </DisplayAtBreakpoint>
  ))
  .add('Display at small, medium or XLarge only', () => (
    <DisplayAtBreakpoint
      screenSmall
      screenMedium
      screenXLarge
    >
      <TestComponent text="Screen is small, medium or XLarge" />
    </DisplayAtBreakpoint>
  ));
