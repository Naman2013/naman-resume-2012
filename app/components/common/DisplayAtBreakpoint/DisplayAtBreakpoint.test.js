import React from 'react';
import { shallow } from 'enzyme';
import DisplayAtBreakpoint from './DisplayAtBreakpoint';

describe('DisplayAtBreakpoint', () => {
  const wrapper = shallow(<DisplayAtBreakpoint><div /></DisplayAtBreakpoint>);
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
