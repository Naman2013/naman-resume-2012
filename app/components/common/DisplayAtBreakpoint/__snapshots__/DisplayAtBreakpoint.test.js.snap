// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayAtBreakpoint should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DisplayAtBreakpoint
    screenLarge={false}
    screenMedium={false}
    screenSmall={false}
    screenXLarge={false}
  >
    <div />
  </DisplayAtBreakpoint>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div />,
        <JSXStyle
          css={
            Array [
              ".root.jsx-1528521221{display:none;width:100%;}",
              ".root.screenSmall.jsx-1528521221{display:block;}",
              "@media screen and (min-width: 768px){.root.screenSmall.jsx-1528521221{display:none;}.root.screenMedium.jsx-1528521221{display:block;}}",
              "@media screen and (min-width: 1024px){.root.screenSmall.jsx-1528521221{display:none;}.root.screenMedium.jsx-1528521221{display:none;}.root.screenLarge.jsx-1528521221{display:block;}}",
              "@media screen and (min-width: 1366px){.root.screenSmall.jsx-1528521221{display:none;}.root.screenMedium.jsx-1528521221{display:none;}.root.screenLarge.jsx-1528521221{display:none;}.root.screenXLarge.jsx-1528521221{display:block;}}",
            ]
          }
          styleId="1528521221"
        />,
      ],
      "className": "jsx-1528521221 root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            ".root.jsx-1528521221{display:none;width:100%;}",
            ".root.screenSmall.jsx-1528521221{display:block;}",
            "@media screen and (min-width: 768px){.root.screenSmall.jsx-1528521221{display:none;}.root.screenMedium.jsx-1528521221{display:block;}}",
            "@media screen and (min-width: 1024px){.root.screenSmall.jsx-1528521221{display:none;}.root.screenMedium.jsx-1528521221{display:none;}.root.screenLarge.jsx-1528521221{display:block;}}",
            "@media screen and (min-width: 1366px){.root.screenSmall.jsx-1528521221{display:none;}.root.screenMedium.jsx-1528521221{display:none;}.root.screenLarge.jsx-1528521221{display:none;}.root.screenXLarge.jsx-1528521221{display:block;}}",
          ],
          "styleId": "1528521221",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div />,
          <JSXStyle
            css={
              Array [
                ".root.jsx-1528521221{display:none;width:100%;}",
                ".root.screenSmall.jsx-1528521221{display:block;}",
                "@media screen and (min-width: 768px){.root.screenSmall.jsx-1528521221{display:none;}.root.screenMedium.jsx-1528521221{display:block;}}",
                "@media screen and (min-width: 1024px){.root.screenSmall.jsx-1528521221{display:none;}.root.screenMedium.jsx-1528521221{display:none;}.root.screenLarge.jsx-1528521221{display:block;}}",
                "@media screen and (min-width: 1366px){.root.screenSmall.jsx-1528521221{display:none;}.root.screenMedium.jsx-1528521221{display:none;}.root.screenLarge.jsx-1528521221{display:none;}.root.screenXLarge.jsx-1528521221{display:block;}}",
              ]
            }
            styleId="1528521221"
          />,
        ],
        "className": "jsx-1528521221 root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              ".root.jsx-1528521221{display:none;width:100%;}",
              ".root.screenSmall.jsx-1528521221{display:block;}",
              "@media screen and (min-width: 768px){.root.screenSmall.jsx-1528521221{display:none;}.root.screenMedium.jsx-1528521221{display:block;}}",
              "@media screen and (min-width: 1024px){.root.screenSmall.jsx-1528521221{display:none;}.root.screenMedium.jsx-1528521221{display:none;}.root.screenLarge.jsx-1528521221{display:block;}}",
              "@media screen and (min-width: 1366px){.root.screenSmall.jsx-1528521221{display:none;}.root.screenMedium.jsx-1528521221{display:none;}.root.screenLarge.jsx-1528521221{display:none;}.root.screenXLarge.jsx-1528521221{display:block;}}",
            ],
            "styleId": "1528521221",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
